From 3a3254e41bc327274730ff723181ddfbdcf91bec Mon Sep 17 00:00:00 2001
From: Huang Chaofan <chaofan.huang@nxp.com>
Date: Tue, 10 Jul 2018 15:50:09 +0800
Subject: [PATCH 4138/5242] MLK-18831 VPU: Adjust the condition for request
 frame when do seek

commit  9e6f76e73f310b7ca82e62b365bc908bd11b8f11 from
https://source.codeaurora.org/external/imx/linux-imx.git

Adjust the condition for request frame when do seek

Signed-off-by: Huang Chaofan <chaofan.huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/mxc/vpu-decoder-b0/vpu_b0.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/mxc/vpu-decoder-b0/vpu_b0.c b/drivers/mxc/vpu-decoder-b0/vpu_b0.c
index 88f2e4a..f8df690a 100644
--- a/drivers/mxc/vpu-decoder-b0/vpu_b0.c
+++ b/drivers/mxc/vpu-decoder-b0/vpu_b0.c
@@ -1646,7 +1646,7 @@ static void vpu_api_event_handler(struct vpu_ctx *ctx, u_int32 uStrIdx, u_int32
 				break;
 			}
 
-			if (!list_empty(&This->drv_q)) {
+			if (!list_empty(&This->drv_q) && !ctx->wait_rst_done) {
 				down(&This->drv_q_lock);
 				list_for_each_entry_safe(p_data_req, p_temp, &This->drv_q, list) {
 					if (p_data_req->status == FRAME_ALLOC
-- 
1.7.9.5

